relay-event-normalization/src/normalize/span/description/sql/parser.rs

```diff
@@ -344,3 +381,62 @@ impl Dialect for DialectWithParameters {
         self.0.parse_statement(parser)
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use sqlparser::parser::Parser;
+
+    use super::*;
+
+    #[test]
+    fn depth() {
+        let query = "SELECT (a + b + (c1 + (c2 + c3)) + c)";
+        let mut parsed = Parser::parse_sql(&GenericDialect {}, query).unwrap();
+        let mut visitor = NormalizeVisitor::new();
+        parsed.visit(&mut visitor);
+        assert_eq!(visitor.max_depth, 8);
+    }
+
+    #[test]
+    fn depth_nested_from() {
+        let query = "SELECT * FROM (SELECT 1)";
+        let mut parsed = Parser::parse_sql(&GenericDialect {}, query).unwrap();
+        let mut visitor = NormalizeVisitor::new();
+        parsed.visit(&mut visitor);
+        assert_eq!(visitor.max_depth, 1);
+    }
+
+    #[test]
+    fn depth_nested_from_2() {
+        let query = "SELECT * FROM (SELECT * FROM (SELECT 1))";
+        let mut parsed = Parser::parse_sql(&GenericDialect {}, query).unwrap();
+        let mut visitor = NormalizeVisitor::new();
+        parsed.visit(&mut visitor);
+        assert_eq!(visitor.max_depth, 2);
```

Comment:
This seems off-by-one, but it should give us an approximation that is good enough to deny heavily nested queries.

Reply:
This should be `3` here? If we count the number overall queries?  But it says `2` - maybe because only 2 of them are nested?  
Asking because want to understand how this works.

Reply:
I simplified the PR to measure only the level of nesting of expressions instead of trying to partially cover other compositions as well.